packages/api/src/hooks/useSortedMT5Accounts.ts

```diff
@@ -0,0 +1,73 @@
+import { useMemo } from 'react';
+import useMT5AccountsList from './useMT5AccountsList';
+import useAvailableMT5Accounts from './useAvailableMT5Accounts';
+
+const useSortedMT5Accounts = () => {
+    const { data: all_available_mt5_accounts } = useAvailableMT5Accounts();
+    const { data: mt5_accounts, ...rest } = useMT5AccountsList();
+
+    const modified_data = useMemo(() => {
+        if (!all_available_mt5_accounts || !mt5_accounts) return;
+
+        return all_available_mt5_accounts?.map(available_account => {
+            const created_account = mt5_accounts?.find(account => {
+                return (
+                    available_account.market_type === account.market_type &&
+                    available_account.shortcode === account.landing_company_short
+                );
+            });
+            if (created_account)
+                return {
+                    ...created_account,
+                    /** Determine if the account is added or not */
+                    is_added: true,
+                } as const;
+
+            return {
+                ...available_account,
+                /** Determine if the account is added or not */
+                is_added: false,
+            } as const;
```

Comment:
```suggestion
            const created_account = mt5_accounts?.find(account => {
                is_added: available_account.market_type === account.market_type &&
                    available_account.shortcode === account.landing_company_short,
```
Can be simplified a lot like this